src/targets/gpu/hiprtc/main.cpp

```diff
@@ -31,10 +31,31 @@
 #include <iostream>
 #include <cstring>
 
+#ifdef _WIN32
+// cppcheck-suppress definePrefix
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#endif
+
 std::vector<char> read_stdin()
 {
     std::vector<char> result;
+#ifdef _WIN32
+    HANDLE std_in = GetStdHandle(STD_INPUT_HANDLE);
+    if(std_in == INVALID_HANDLE_VALUE)
+        MIGRAPHX_THROW("STDIN invalid handle (" + std::to_string(GetLastError()) + ")");
+    constexpr std::size_t BUFFER_SIZE = 1024;
+    DWORD bytes_read;
+    TCHAR buffer[BUFFER_SIZE];
+    for(;;)
+    {
+        BOOL status = ReadFile(std_in, buffer, BUFFER_SIZE, &bytes_read, nullptr);
+        if(status == FALSE or bytes_read == 0)
+            break;
 
+        result.insert(result.end(), buffer, buffer + bytes_read);
+    }
+#else
```

Comment:
Why is this needed? The below code should work on windows.

Reply:
It is not working. IPC on Windows differs from Linux. Pipes on Windows are sections of shared memory and not artificial I/O channels. The only way to make it work is to use `_pipe` or `_popen` in MIGraphX, which emulates POSIX functions on Windows. However, as we already discussed, `_pipe`, `_popen`, etc., have severe limitations and only can be used in console applications. However, most use cases are Windows programs, Windows Runtime applications, etc. According to the MSFT documentation, in such a scenario, the functions return an invalid file pointer that causes the program to stop responding indefinitely. This is what we are experiencing here, for example, when executing `test_gpu_quantization`.

Reply:
> IPC is not part of the C/C++ standard but is OS-specific based on the OS standard implemented. 

This is just using `stdin` which is part of standard C/C++. Why does`stdin` not read from `GetStdHandle(STD_INPUT_HANDLE)`? From the documentation [here](https://learn.microsoft.com/en-us/windows/console/getstdhandle) they look like they are reading from the same place. Is there another documentation that explains why and what `stdin` actually reads?

> Pipes on Windows are sections of shared memory and not artificial I/O channels.

You reading from it like a file with `ReadFile`, so dont understand why this implementation detail matters. Furthermore, `FILE*` or iostreams can read from memory buffers not just files as well. 

> The only way to make it work is to use _pipe or _popen

Why doesnt `CreatePipe` work?

Also curious why you dont pass `CREATE_NO_WINDOW` to `CreateProcess` as well? I would assume this would open a console window in GUI applications(which we dont want).

Reply:
> have severe limitations and only can be used in console applications (require command shell support).

So from what I am reading [here](https://learn.microsoft.com/en-us/windows/console/console-handles), [`GetStdHandle`](https://learn.microsoft.com/en-us/windows/console/getstdhandle) can only be used from a console application. So to make this work we would need to launch this as a console application anyways, which is why we would need `CREATE_NO_WINDOW` to prevent a console popup happening in a GUI application.

I still dont understand why `stdin` doesn't work for this case.

Reply:
The `CREATE_NO_WINDOWS` cannot be used when the parent process has captured the `stdin` or `stdout` of a child process. This is the case here, and the windows will not appear because both streams have been captured.

Reply:
It turned out the mode of `stdin` was `_O_TEXT`, and since Message Pack generates binary data, there were errors in data transmission between processes. When I switched `stdin` to `_O_BINARY` mode, everything started to work. So, I removed Win32-specific code from the HIPRTC driver.
Because Windows 'end-of-line' comprises two characters (\r\n), it is essential to differentiate between text and binary when sending data.

Reply:
I added the unit test for the 'process' class in this PR.

Reply:
I am composing an email regarding `tiny-process-library`.

Reply:
Additionally, I have separated process arguments from a module (path to executable) and added support for environmental variables.